Tasks/DeployAzureResourceGroup/MachineGroupAgentExtensionManager.ts

```diff
@@ -0,0 +1,179 @@
+var computeManagementClient = require("azure-arm-compute");
+import util = require("util");
+import tl = require("vsts-task-lib/task");
+import azure_utils = require("./AzureUtil");
+import deployAzureRG = require("./DeployAzureRG");
+import Q = require("q");
+import constants = require("./Constants");
+
+export class MachineGroupAgentExtensionManager {
+    private taskParameters: deployAzureRG.AzureRGTaskParameters;
+    private credentials;
+    private subscriptionId: string;
+    private azureUtils;
+    private failureCount: number;
+    private successCount: number;
+    private errors: string;
+    private vmCount: number;
+    private deferred;
+    private operation: string;
+    private computeClient;
+
+    constructor(taskParameters: deployAzureRG.AzureRGTaskParameters) {
+        this.taskParameters = taskParameters;
+        this.credentials = this.taskParameters.credentials;
+        this.subscriptionId = this.taskParameters.subscriptionId;
+        this.azureUtils = new azure_utils.AzureUtil(this.taskParameters);
+        this.successCount = 0;
+        this.failureCount = 0;
```

Comment:
please consider a design where you don't have to instantiate this class everytime you want to call a method.